### ISSUE TITLE
A compile bug in for loop when let to var

### ISSUE THREAD

**Issue by zhmushan** [Microsoft/TypeScript#26570]
**TypeScript Version:**  3.1.0-dev.201xxxxx

**Search Terms:**
for loop, var, let, es5

**Code**
```ts
for(
    let i = (setTimeout(() => console.log(i), 0), 0);
    i < 2;
    i++
) { }
```

**Expected behavior:**
```
> tsc test.ts -t es6 --outFile test.es6.out && node test.es6.out
< 0
> tsc test.ts -t es5 --outFile test.es5.out && node test.es5.out
< 0
```

**Actual behavior:**
```
> tsc test.ts -t es6 --outFile test.es6.out && node test.es6.out
< 0
> tsc test.ts -t es5 --outFile test.es5.out && node test.es5.out
< 2
```

**Comment by zhmushan** [comment]
**Expected compilation result:**
```
var _loop_1 = function (i) {
};
for (
    var i = (setTimeout(function () { return console.log(i); }, 0), 0),
    _i = i;
    _i < 2;
    _i++
) {
    _loop_1(_i);
}
```

**Actual compilation result:**
```
var _loop_1 = function (i) {
};
for (var i = (setTimeout(function () { return console.log(i); }, 0), 0); i < 2; i++) {
    _loop_1(i);
}
```

**Comment by kpreisser** [comment]
I think this is a duplicate of #17632.

**Comment by rbuckton** [comment]
Note that there is a fix for #17632 under review in the PR #27204.
